src/app/store/manage-tokens/manage-tokens.hooks.ts

```diff
@@ -0,0 +1,17 @@
+import { useDispatch } from 'react-redux';
+
+import { useCurrentAccountIndex } from '../accounts/account';
+import { addToken } from './manage-tokens.slice';
+
+// loads tokens to store if it doesnt exist & sets enabled as false
+function useInitTokensAsDisabled({ ids }: { ids: string[] }) {
```

Comment:
I think there might be a simpler implementation for the store, that doesn't pre-setting of state. 

Conceptually, we have two sets of binary states. Our defaults, supported/unsupported, and the new ones you've added, a user set enabled/disabled.

Suggest implementing this in such a way where, by default, the `manageTokens` slice is empty, as users have made no state changes. An `undefined` for a look up of the token in the manageTokens slice means the user as set no preference, fallback to default. 

Only when modifying a rule, would we then save the state. I think this way we could avoid the initial "set all the state" action.